chains/ethereum/server/src/lib.rs

```diff
@@ -308,11 +308,22 @@ mod tests {
             wallet.eth_send_call(contract_address.0, call.abi_encode(), 0).await?
         };
         let receipt = wallet.eth_transaction_receipt(tx_hash).await?.unwrap();
-        assert_eq!(receipt.logs.len(), 1);
+        let assert_result = std::panic::catch_unwind(|| {
+            assert_eq!(receipt.logs.len(), 1);
+        });
```

Comment:
`std::panic::catch_unwind` doesn't support async methods, the idea is wrap the whole test into an async block, and execute the cleanup step regardless if the async block panics or not.